<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1140</width>
    <height>573</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Bitmap to processing path</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="LB_ImageViewer" name="graphicResult">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>12</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="LB_ImageViewer" name="graphicSource">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>12</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="label_imgInfoResult">
      <property name="text">
       <string>Image not open</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_imgInfoSource">
      <property name="text">
       <string>Image not open</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget_undo">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>245</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>200</width>
     <height>524287</height>
    </size>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Operation Record</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_arguments">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>230</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>200</width>
     <height>230</height>
    </size>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Arguments</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="2" colspan="2">
      <widget class="QComboBox" name="comboBox_scaleFactor">
       <property name="currentIndex">
        <number>5</number>
       </property>
       <item>
        <property name="text">
         <string>50%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>67%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>75%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>80%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>90%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>100%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>110%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>125%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>150%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>175%</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>200%</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="2" column="0" colspan="3">
      <widget class="QLabel" name="label_minPathLen">
       <property name="text">
        <string>Minimum edge length:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_ahpha">
       <property name="text">
        <string>Sharpness coefficient:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_colinearTol">
       <property name="text">
        <string>Colinear tolerance:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2" colspan="2">
      <widget class="QSpinBox" name="spinBox_threshold">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>2</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="4">
      <widget class="QCheckBox" name="checkBox_showVertex">
       <property name="text">
        <string>Show vertex</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="4" column="3">
      <widget class="QSpinBox" name="spinBox_bezierStep">
       <property name="toolTip">
        <string>Number of smooth points replacing the original contour</string>
       </property>
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>15</number>
       </property>
       <property name="value">
        <number>3</number>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="QLabel" name="label_bezierStep">
       <property name="text">
        <string>Interpolation count:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="3">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_colinearTol">
       <property name="toolTip">
        <string>In the vectorization process, pixels are in the maximum tolerance of a straight line</string>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QSpinBox" name="spinBox_minPathLen">
       <property name="toolTip">
        <string>The minimum number of pixels of an outline</string>
       </property>
       <property name="maximum">
        <number>999</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="4">
      <widget class="QCheckBox" name="checkBox_showContours">
       <property name="text">
        <string>Show contours</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="label_threshold">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Binarization threshold:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_scaleFactor">
       <property name="text">
        <string>Scale factor:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1" colspan="3">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_alpha">
       <property name="toolTip">
        <string>The larger the sharp coefficient, the more smooth corners of the image contour</string>
       </property>
       <property name="singleStep">
        <double>0.050000000000000</double>
       </property>
       <property name="value">
        <double>0.550000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_openImg"/>
   <addaction name="action_saveAsDXF"/>
   <addaction name="action_saveAsImg"/>
   <addaction name="separator"/>
   <addaction name="action_back"/>
   <addaction name="action_next"/>
   <addaction name="action_resetOperation"/>
   <addaction name="separator"/>
   <addaction name="action_filter"/>
   <addaction name="action_findContours"/>
   <addaction name="action_binary"/>
   <addaction name="action_findThreshold"/>
   <addaction name="separator"/>
   <addaction name="action_generateResult"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1140</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_openImg"/>
    <addaction name="action_saveAsImg"/>
    <addaction name="action_saveAsDXF"/>
   </widget>
   <widget class="QMenu" name="menuOperation">
    <property name="title">
     <string>Operation</string>
    </property>
    <addaction name="action_resetOperation"/>
    <addaction name="separator"/>
    <addaction name="action_back"/>
    <addaction name="action_next"/>
    <addaction name="separator"/>
    <addaction name="action_gray"/>
    <addaction name="action_binary"/>
    <addaction name="action_sharpen"/>
    <addaction name="action_filter"/>
    <addaction name="action_thinning"/>
    <addaction name="action_findThreshold"/>
    <addaction name="action_findContours"/>
    <addaction name="separator"/>
    <addaction name="action_cannyContours"/>
    <addaction name="action_sobelContours"/>
   </widget>
   <widget class="QMenu" name="menuVectorization">
    <property name="title">
     <string>Vectorization</string>
    </property>
    <addaction name="action_generateResult"/>
    <addaction name="separator"/>
    <addaction name="action_houghLine"/>
    <addaction name="action_houghCircle"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuOperation"/>
   <addaction name="menuVectorization"/>
  </widget>
  <action name="action_openImg">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_open.png</normaloff>:/icons/img_open.png</iconset>
   </property>
   <property name="text">
    <string>Open Image</string>
   </property>
   <property name="statusTip">
    <string>Open Image</string>
   </property>
  </action>
  <action name="action_generateResult">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_generate.png</normaloff>:/icons/img_generate.png</iconset>
   </property>
   <property name="text">
    <string>Generate results</string>
   </property>
   <property name="statusTip">
    <string>Generate results</string>
   </property>
  </action>
  <action name="action_back">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_back.png</normaloff>:/icons/img_back.png</iconset>
   </property>
   <property name="text">
    <string>Last step</string>
   </property>
   <property name="statusTip">
    <string>Last step</string>
   </property>
  </action>
  <action name="action_next">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_next.png</normaloff>:/icons/img_next.png</iconset>
   </property>
   <property name="text">
    <string>Next step</string>
   </property>
   <property name="statusTip">
    <string>Next step</string>
   </property>
  </action>
  <action name="action_filter">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_filter.png</normaloff>:/icons/img_filter.png</iconset>
   </property>
   <property name="text">
    <string>Median filter</string>
   </property>
   <property name="statusTip">
    <string>Median filter</string>
   </property>
  </action>
  <action name="action_findContours">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_edge.png</normaloff>:/icons/img_edge.png</iconset>
   </property>
   <property name="text">
    <string>Find contours</string>
   </property>
   <property name="statusTip">
    <string>Find contours</string>
   </property>
  </action>
  <action name="action_binary">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_binary.png</normaloff>:/icons/img_binary.png</iconset>
   </property>
   <property name="text">
    <string>Binary</string>
   </property>
   <property name="statusTip">
    <string>Binary</string>
   </property>
  </action>
  <action name="action_findThreshold">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_calculateTh.png</normaloff>:/icons/img_calculateTh.png</iconset>
   </property>
   <property name="text">
    <string>Best threshold</string>
   </property>
   <property name="statusTip">
    <string>Best threshold</string>
   </property>
  </action>
  <action name="action_saveAsImg">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_saveImg.png</normaloff>:/icons/img_saveImg.png</iconset>
   </property>
   <property name="text">
    <string>Save as image</string>
   </property>
   <property name="toolTip">
    <string>Save as image</string>
   </property>
   <property name="statusTip">
    <string>Save as image</string>
   </property>
  </action>
  <action name="action_saveAsDXF">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_saveDXF.png</normaloff>:/icons/img_saveDXF.png</iconset>
   </property>
   <property name="text">
    <string>Save as dxf</string>
   </property>
   <property name="toolTip">
    <string>Save as dxf</string>
   </property>
   <property name="statusTip">
    <string>Save as dxf</string>
   </property>
  </action>
  <action name="action_cannyContours">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_cannyEdge.png</normaloff>:/icons/img_cannyEdge.png</iconset>
   </property>
   <property name="text">
    <string>Canny Contours</string>
   </property>
   <property name="toolTip">
    <string>Canny Contours</string>
   </property>
   <property name="statusTip">
    <string>Canny Contours</string>
   </property>
  </action>
  <action name="action_sobelContours">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_sobelEdge.png</normaloff>:/icons/img_sobelEdge.png</iconset>
   </property>
   <property name="text">
    <string>Sobel Contours</string>
   </property>
   <property name="toolTip">
    <string>Sobel Contours</string>
   </property>
   <property name="statusTip">
    <string>Sobel Contours</string>
   </property>
  </action>
  <action name="action_gray">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_gray.png</normaloff>:/icons/img_gray.png</iconset>
   </property>
   <property name="text">
    <string>Gray</string>
   </property>
   <property name="toolTip">
    <string>Gray</string>
   </property>
   <property name="statusTip">
    <string>Gray</string>
   </property>
  </action>
  <action name="action_sharpen">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_sharpen.png</normaloff>:/icons/img_sharpen.png</iconset>
   </property>
   <property name="text">
    <string>Sharpen</string>
   </property>
   <property name="toolTip">
    <string>Sharpen</string>
   </property>
   <property name="statusTip">
    <string>Sharpen</string>
   </property>
  </action>
  <action name="action_thinning">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_thinning.png</normaloff>:/icons/img_thinning.png</iconset>
   </property>
   <property name="text">
    <string>Thinning</string>
   </property>
   <property name="toolTip">
    <string>Thinning</string>
   </property>
   <property name="statusTip">
    <string>Thinning</string>
   </property>
  </action>
  <action name="action_houghLine">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/hough_line.png</normaloff>:/icons/hough_line.png</iconset>
   </property>
   <property name="text">
    <string>Detect Segment</string>
   </property>
   <property name="toolTip">
    <string>Detect Segment</string>
   </property>
   <property name="statusTip">
    <string>Detect Segment</string>
   </property>
  </action>
  <action name="action_houghCircle">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/hough_circle.png</normaloff>:/icons/hough_circle.png</iconset>
   </property>
   <property name="text">
    <string>Detect Circle</string>
   </property>
   <property name="toolTip">
    <string>Detect Circle</string>
   </property>
   <property name="statusTip">
    <string>Detect Circle</string>
   </property>
  </action>
  <action name="action_resetOperation">
   <property name="icon">
    <iconset resource="ImageAlgo.qrc">
     <normaloff>:/icons/img_resetOperation.png</normaloff>:/icons/img_resetOperation.png</iconset>
   </property>
   <property name="text">
    <string>Reset Operation</string>
   </property>
   <property name="toolTip">
    <string>Reset Operation</string>
   </property>
   <property name="statusTip">
    <string>Reset Operation</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LB_ImageViewer</class>
   <extends>QGraphicsView</extends>
   <header>LB_Image/LB_ImageViewer.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="ImageAlgo.qrc"/>
 </resources>
 <connections/>
</ui>
